{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "0442f9c2",
   "metadata": {},
   "source": [
    "# Stock Purchases Thoughts\n",
    "\n",
    "**12/05/2020**\n",
    "If you have no idea what you are doing, rely on charts and TA, it can tell you a good time to sell.\n",
    "\n",
    "Never underestimate FOMO\n",
    "\n",
    "**11/11/2020**\n",
    "\n",
    "* MTRX loop insights - spiked to 0.50 - shouldn't have sold\n",
    "* DCM - epic report good buy\n",
    "* PKK - went down a bunch still a decent buy with a zack report at 3.8 USD - 5.0 CAD\n",
    "* ART - down should be knee jerk reaction\n",
    "* BEE - should recover with other pandemic stocks  looking promising 100% retention must have good product and good peope\n",
    "* CGX - sold cause movie theatre - day trade\n",
    "\n",
    "**11/09/2020**\n",
    "\n",
    "MTRX loop insights video mentioning pfizer, why the heck did I sell it. Was this a good idea? probably not.\n",
    "\n",
    "Bought DCM - should be up 10x and peak should be up 10 x.\n",
    "\n",
    "Diversification - perhaps should have just spent more money instead.\n",
    "\n",
    "Split advice by year eventually\n",
    "**10/28/2020**\n",
    "\n",
    "Stock market crashing dont feel like selling\n",
    "\n",
    "- CMC.CN - Needs to hit targets and shit the facilities built\n",
    "- PKK.CN - main focus on china not covid center spot\n",
    "- IDK.CN - nav of 0.41 usually stocks trade 2x nav and pkk.cn + mtrx doing well this year\n",
    "- MTRX.CN - contact tracing, if adopted should do well\n",
    "- ART.V - business does well in covid19\n",
    "\n",
    "Can trim some loop on the uptick 10k shares, same with cielo, 40k shares\n",
    "\n",
    "**10/23/2020**\n",
    "Peak to aphria\n",
    "\n",
    "shouldn't have fucking sold\n",
    "\n",
    "Peak still growing\n",
    "\n",
    "Pot not tech, can't grow fast\n",
    "aphria thoughts, if biden wins, harris pot decriminalization in america, temp surge, dump aphria, buy arht?\n",
    "\n",
    "**10/11/2020**\n",
    "\n",
    "Doubling down on peak as some guy in ceo.ca/pkk posted an article about china growing. They have control of the virus, it makes sense to invest in a country that is growing that is not in the state, espeically if they believe they are severely undervalued with high margins and the opportunity to give a dividend.\n",
    "\n",
    "In addition, sheldon has a position in pkk.\n",
    "\n",
    "Cielo seems to be doing well, if production numbers are 1000 L/h (pretty high), I would expect 100x return sometime in the future. After all, they have done the work to start growing like crazy as plastic production is projected to increase and there is no solution as good as cielo at the moment.\n",
    "\n",
    "**09/13/2020**\n",
    "KUU MIGHT BEINSOVLENT\n",
    "Wait on BYL for 5G\n",
    "\n",
    "Think about VIS.V if it makes 334,250 as projected in Q4\n",
    "\n",
    "The stocks I looked at recently that were interesting as of 09/15/2020\n",
    "\n",
    "| Ticker   | Long Name                       |\n",
    "| :---     | ---:                            |\n",
    "| PAI      | Predictiv AI Inc.               |\n",
    "| RW       | RenoWorks Software Inc          |\n",
    "| DFT      | Dimension Five Technologies Inc |\n",
    "| KUU      | Kuuhubb Inc                     |\n",
    "| BYL      | Baylin Technology Inc           |\n",
    "| CMC      | Cielo Waste Solutions Inc       |\n",
    "\n",
    "\n",
    "**09/08/2020**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b8a63bf8",
   "metadata": {},
   "outputs": [
    {
     "ename": "OperationalError",
     "evalue": "unable to open database file",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mOperationalError\u001b[0m                          Traceback (most recent call last)",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/peewee.py:3192\u001b[0m, in \u001b[0;36mDatabase.connect\u001b[0;34m(self, reuse_if_open)\u001b[0m\n\u001b[1;32m   3191\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m __exception_wrapper__:\n\u001b[0;32m-> 3192\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_state\u001b[38;5;241m.\u001b[39mset_connection(\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_connect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m)\n\u001b[1;32m   3193\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mserver_version \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/peewee.py:3536\u001b[0m, in \u001b[0;36mSqliteDatabase._connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   3535\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m ImproperlyConfigured(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSQLite driver not installed!\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m-> 3536\u001b[0m conn \u001b[38;5;241m=\u001b[39m \u001b[43msqlite3\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_timeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3537\u001b[0m \u001b[43m                       \u001b[49m\u001b[43misolation_level\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect_params\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3538\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n",
      "\u001b[0;31mOperationalError\u001b[0m: unable to open database file",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mOperationalError\u001b[0m                          Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[1], line 7\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mmplfinance\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mmpf\u001b[39;00m\n\u001b[1;32m      6\u001b[0m vst \u001b[38;5;241m=\u001b[39m yf\u001b[38;5;241m.\u001b[39mTicker(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mVST.CN\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m----> 7\u001b[0m data \u001b[38;5;241m=\u001b[39m \u001b[43mvst\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mhistory\u001b[49m\u001b[43m(\u001b[49m\u001b[43minterval\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m1d\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mstart\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m2020-08-08\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mend\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m2020-09-10\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\n\u001b[1;32m      8\u001b[0m mpf\u001b[38;5;241m.\u001b[39mplot(\n\u001b[1;32m      9\u001b[0m     data,\n\u001b[1;32m     10\u001b[0m     \u001b[38;5;28mtype\u001b[39m\u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mcandle\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m   (...)\u001b[0m\n\u001b[1;32m     15\u001b[0m     ),\n\u001b[1;32m     16\u001b[0m )\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/yfinance/utils.py:114\u001b[0m, in \u001b[0;36mlog_indent_decorator.<locals>.wrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    111\u001b[0m logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mEntering \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfunc\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m()\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m    113\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m IndentationContext():\n\u001b[0;32m--> 114\u001b[0m     result \u001b[38;5;241m=\u001b[39m \u001b[43mfunc\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    116\u001b[0m logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mExiting \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mfunc\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m()\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m    117\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m result\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/yfinance/base.py:148\u001b[0m, in \u001b[0;36mTickerBase.history\u001b[0;34m(self, period, interval, start, end, prepost, actions, auto_adjust, back_adjust, repair, keepna, proxy, rounding, timeout, debug, raise_errors)\u001b[0m\n\u001b[1;32m    145\u001b[0m end_user \u001b[38;5;241m=\u001b[39m end\n\u001b[1;32m    146\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m start \u001b[38;5;129;01mor\u001b[39;00m period \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;129;01mor\u001b[39;00m period\u001b[38;5;241m.\u001b[39mlower() \u001b[38;5;241m==\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mmax\u001b[39m\u001b[38;5;124m\"\u001b[39m:\n\u001b[1;32m    147\u001b[0m     \u001b[38;5;66;03m# Check can get TZ. Fail => probably delisted\u001b[39;00m\n\u001b[0;32m--> 148\u001b[0m     tz \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_get_ticker_tz\u001b[49m\u001b[43m(\u001b[49m\u001b[43mproxy\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    149\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m tz \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    150\u001b[0m         \u001b[38;5;66;03m# Every valid ticker has a timezone. Missing = problem\u001b[39;00m\n\u001b[1;32m    151\u001b[0m         err_msg \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mNo timezone found, symbol may be delisted\u001b[39m\u001b[38;5;124m\"\u001b[39m\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/yfinance/base.py:1644\u001b[0m, in \u001b[0;36mTickerBase._get_ticker_tz\u001b[0;34m(self, proxy, timeout)\u001b[0m\n\u001b[1;32m   1642\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_tz \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   1643\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_tz\n\u001b[0;32m-> 1644\u001b[0m cache \u001b[38;5;241m=\u001b[39m \u001b[43mutils\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget_tz_cache\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   1645\u001b[0m tz \u001b[38;5;241m=\u001b[39m cache\u001b[38;5;241m.\u001b[39mlookup(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mticker)\n\u001b[1;32m   1647\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m tz \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m utils\u001b[38;5;241m.\u001b[39mis_valid_timezone(tz):\n\u001b[1;32m   1648\u001b[0m     \u001b[38;5;66;03m# Clear from cache and force re-fetch\u001b[39;00m\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/yfinance/utils.py:985\u001b[0m, in \u001b[0;36mget_tz_cache\u001b[0;34m()\u001b[0m\n\u001b[1;32m    983\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m _tz_cache \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m    984\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[0;32m--> 985\u001b[0m         _tz_cache \u001b[38;5;241m=\u001b[39m \u001b[43m_TzCache\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    986\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m _TzCacheException \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[1;32m    987\u001b[0m         get_yf_logger()\u001b[38;5;241m.\u001b[39minfo(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mFailed to create TzCache, reason: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00merr\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    988\u001b[0m                              \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mTzCache will not be used. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m    989\u001b[0m                              \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mTip: You can direct cache to use a different location with \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mset_tz_cache_location(mylocation)\u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/yfinance/utils.py:926\u001b[0m, in \u001b[0;36m_TzCache.__init__\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    925\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21m__init__\u001b[39m(\u001b[38;5;28mself\u001b[39m):\n\u001b[0;32m--> 926\u001b[0m     \u001b[43mdb\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m    927\u001b[0m     db\u001b[38;5;241m.\u001b[39mcreate_tables([KV])\n\u001b[1;32m    929\u001b[0m     old_cache_file_path \u001b[38;5;241m=\u001b[39m _os\u001b[38;5;241m.\u001b[39mpath\u001b[38;5;241m.\u001b[39mjoin(_cache_dir, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtkr-tz.csv\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/peewee.py:3191\u001b[0m, in \u001b[0;36mDatabase.connect\u001b[0;34m(self, reuse_if_open)\u001b[0m\n\u001b[1;32m   3188\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m OperationalError(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mConnection already opened.\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[1;32m   3190\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_state\u001b[38;5;241m.\u001b[39mreset()\n\u001b[0;32m-> 3191\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m __exception_wrapper__:\n\u001b[1;32m   3192\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_state\u001b[38;5;241m.\u001b[39mset_connection(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_connect())\n\u001b[1;32m   3193\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mserver_version \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/peewee.py:3019\u001b[0m, in \u001b[0;36mExceptionWrapper.__exit__\u001b[0;34m(self, exc_type, exc_value, traceback)\u001b[0m\n\u001b[1;32m   3017\u001b[0m new_type \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mexceptions[exc_type\u001b[38;5;241m.\u001b[39m\u001b[38;5;18m__name__\u001b[39m]\n\u001b[1;32m   3018\u001b[0m exc_args \u001b[38;5;241m=\u001b[39m exc_value\u001b[38;5;241m.\u001b[39margs\n\u001b[0;32m-> 3019\u001b[0m \u001b[43mreraise\u001b[49m\u001b[43m(\u001b[49m\u001b[43mnew_type\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mnew_type\u001b[49m\u001b[43m(\u001b[49m\u001b[43mexc_value\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[43mexc_args\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtraceback\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/peewee.py:192\u001b[0m, in \u001b[0;36mreraise\u001b[0;34m(tp, value, tb)\u001b[0m\n\u001b[1;32m    190\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mreraise\u001b[39m(tp, value, tb\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[1;32m    191\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m value\u001b[38;5;241m.\u001b[39m__traceback__ \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m tb:\n\u001b[0;32m--> 192\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m value\u001b[38;5;241m.\u001b[39mwith_traceback(tb)\n\u001b[1;32m    193\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m value\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/peewee.py:3192\u001b[0m, in \u001b[0;36mDatabase.connect\u001b[0;34m(self, reuse_if_open)\u001b[0m\n\u001b[1;32m   3190\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_state\u001b[38;5;241m.\u001b[39mreset()\n\u001b[1;32m   3191\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m __exception_wrapper__:\n\u001b[0;32m-> 3192\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_state\u001b[38;5;241m.\u001b[39mset_connection(\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_connect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m)\n\u001b[1;32m   3193\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mserver_version \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   3194\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_set_server_version(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_state\u001b[38;5;241m.\u001b[39mconn)\n",
      "File \u001b[0;32m/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/peewee.py:3536\u001b[0m, in \u001b[0;36mSqliteDatabase._connect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   3534\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m sqlite3 \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[1;32m   3535\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m ImproperlyConfigured(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSQLite driver not installed!\u001b[39m\u001b[38;5;124m'\u001b[39m)\n\u001b[0;32m-> 3536\u001b[0m conn \u001b[38;5;241m=\u001b[39m \u001b[43msqlite3\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mdatabase\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mtimeout\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_timeout\u001b[49m\u001b[43m,\u001b[49m\n\u001b[1;32m   3537\u001b[0m \u001b[43m                       \u001b[49m\u001b[43misolation_level\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;241;43m*\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconnect_params\u001b[49m\u001b[43m)\u001b[49m\n\u001b[1;32m   3538\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m   3539\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_add_conn_hooks(conn)\n",
      "\u001b[0;31mOperationalError\u001b[0m: unable to open database file"
     ]
    }
   ],
   "source": [
    "import yfinance as yf\n",
    "import matplotlib.pyplot as plt\n",
    "import matplotlib.dates as mdates\n",
    "import mplfinance as mpf\n",
    "\n",
    "vst = yf.Ticker(\"VST.CN\")\n",
    "data = vst.history(interval=\"1d\", start=\"2020-08-08\", end=\"2020-09-10\")\n",
    "mpf.plot(\n",
    "    data,\n",
    "    type=\"candle\",\n",
    "    mav=4,\n",
    "    title=\"Victory Square Purchases Dates\",\n",
    "    vlines=dict(\n",
    "        vlines=[\"2020-08-26\", \"2020-09-03\", \"2020-09-08\"], linewidths=(0.5, 0.5, 0.5)\n",
    "    ),\n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5df40efe",
   "metadata": {},
   "source": [
    "The first few purchases on August 26 and September 03 were profitable, after the failed Sept 08 purchases and surprise drop I emailed the investor relations guy asking about india approval.\n",
    "\n",
    "My brief calculations of the covid-19 test selling well make me think if they sell well (80 \\%), the stock will continue the rise.\n",
    "\n",
    "The combination of covid-19 increasing, the need to know if you were previously infected for health care workers perhaps even vaccines, make me think that the stock will go up in the short term in the next few months.\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "## Cloud Architecture V1\n",
    "\n",
    "Since diagrams added a bunch of nodes, I can now fully diagram my current stock market CI/CD pipelines and tools.\n",
    "\n",
    "To run on colab install graphviz\n",
    "```\n",
    "!pip install graphviz\n",
    "!apt-get install graphviz\n",
    "```\n",
    "\n",
    "Install the diagrams package\n",
    "```\n",
    "!pip install diagrams\n",
    "```\n",
    "\n",
    "```{figure} /_static/diagrams/v1stockarch.png\n",
    "```\n",
    "\n",
    "\n",
    "## Cloud Architecture V0\n",
    "\n",
    "Make this seperate chapter \n",
    "\n",
    "\n",
    "As for sept 28 2020\n",
    "\n",
    "```{figure} /_static/diagrams/v0stockarch.png\n",
    "```\n",
    "\n",
    "my basic cloud archtecture was based on gcp. Primarily using google cloud build for automated deployed and the various serverless compute options. Aimed to stay within free tier but sometimes I paid cash for this (TODO make a blog post about my stocks hosting stuff)."
   ]
  }
 ],
 "metadata": {
  "jupytext": {
   "text_representation": {
    "extension": ".md",
    "format_name": "myst"
   }
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  },
  "source_map": [
   10,
   86,
   103
  ]
 },
 "nbformat": 4,
 "nbformat_minor": 5
}